commands/migratecmd.go

```diff
@@ -250,18 +272,26 @@ func (mr *MigrateRunner) migrateObjs(cmObjs []object.ObjMetadata, reader io.Read
 	if err != nil {
 		return err
 	}
-	// Filter the ConfigMap inventory object.
-	rgInv, err := findResourceGroupInv(objs)
+	inv, objs, err := mr.rgLoader.InventoryInfo(objs)
 	if err != nil {
 		return err
 	}
-	rgInvClient, err := mr.rgProvider.InventoryClient()
-	if err != nil {
+
+	applier := apply.NewApplier(mr.rgProvider)
+
+	if err := applier.Initialize(); err != nil {
 		return err
 	}
-	inv := live.WrapInventoryInfoObj(rgInv)
-	_, err = rgInvClient.Merge(inv, cmObjs)
+	ch := applier.Run(context.TODO(), inv, objs, apply.Options{
```

Comment:
This worries me because it is a major abstraction violation. `kpt` should not know about the lower-level guts of how apply is implemented. I think this points to a bigger problem with the `InventoryPolicy` which Morten has pointed out. The `InventoryPolicy` annotations significantly increase the scope of the inventory mechanism by now including every object that is applied. Before, the algorithm only dealt with one (inventory) object during apply. Now it deals with every object that is applied.

Reply:
How about the `migrate` command only updates the inventory object. It gets the inventory list from the ConfigMap inventory object and then assign it to the ResourceGroup CR. Then the `owning-inventory` is handled by the next `apply` after the migration.

Reply:
So while I agree that kpt knowing about the underlying inventory is not ideal, I don't think it is a big issue for a migration code. The way the InventoryPolicy is designed, kpt will not have to know about it once we have migrated. 

The apply mechanism in general already deals with every resource. This adds an extra annotation to the resources, but otherwise doesn't really affect the core mechanism of the inventory algorithm. The most important way the apply logic is affected is around checking for this annotation, but that seems pretty well in line with how we handle lifecycle directives.

Reply:
We need to update the `owning-inventory` at the same time when migrating the inventory object since the next `apply` can't update the `owning-inventory` object correctly. See https://github.com/GoogleContainerTools/kpt/pull/1307#discussion_r546922602.